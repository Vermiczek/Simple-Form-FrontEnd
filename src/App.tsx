import { CustomForm } from "./CustomForm";
import { StyledApp } from "./styles/App.styled";

function App() {
  return (
    <StyledApp>
      <div className="App">
        <CustomForm />
      </div>
    </StyledApp>
  );
}

export default App;
