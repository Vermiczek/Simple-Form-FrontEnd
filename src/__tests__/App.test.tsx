import React from "react";
import { render, screen } from "@testing-library/react";
import App from "../App";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App />, div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
