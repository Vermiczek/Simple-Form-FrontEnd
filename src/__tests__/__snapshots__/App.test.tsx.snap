// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-hKMtZM dZkCYW"
>
  <div
    className="App"
  >
    <div
      className="sc-dkzDqf jIXyOr"
    >
      <h1>
        Place your order!
      </h1>
      <form
        action="#"
        onReset={[Function]}
        onSubmit={[Function]}
      >
        <div
          className="input-field"
        >
          <label>
            Name
          </label>
          <input
            className="styled-input"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
        </div>
        <div
          className="input-field"
        >
          <label>
            Preparation time
          </label>
          <div
            className="sc-gsnTZi fMjxUu"
          >
            <div
              className="duration-wrapper"
            >
              <input
                className="styled-input-tiny"
                data-testid="input-hour"
                max="99"
                maxLength={1}
                min="0"
                onChange={[Function]}
                type="number"
                value="00"
              />
              <div>
                :
              </div>
              <input
                className="styled-input-tiny"
                data-testid="input-minute"
                max="59"
                min="0"
                onChange={[Function]}
                type="number"
                value="00"
              />
              <div>
                :
              </div>
              <input
                className="styled-input-tiny"
                data-testid="input-second"
                max="59"
                min="0"
                onChange={[Function]}
                type="number"
                value="00"
              />
            </div>
          </div>
        </div>
        <div
          className="input-field"
        >
          <label>
            Dish type
          </label>
          <select
            className="styled-input"
            name="type"
            onBlur={[Function]}
            onChange={[Function]}
            value="pizza"
          >
            <option
              className="option"
              value="pizza"
            >
              Pizza
            </option>
            <option
              className="option"
              value="soup"
            >
              Soup
            </option>
            <option
              className="option"
              value="sandwich"
            >
              Sandwich
            </option>
          </select>
        </div>
        <div
          className="input-field"
        >
          <div
            className="sc-bczRLJ ecqoBE"
          >
            <label>
              Number of slices
            </label>
            <input
              className="styled-input"
              min="0"
              name="no_of_slices"
              onBlur={[Function]}
              onChange={[Function]}
              type="number"
              value={4}
            />
            <label>
              Diameter
            </label>
            <input
              className="styled-input"
              min="0"
              name="diameter"
              onBlur={[Function]}
              onChange={[Function]}
              step="0.01"
              type="number"
              value={50}
            />
          </div>
        </div>
        <button
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Reset
        </button>
        <button
          disabled={false}
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
